### PR TITLE
dolt/{go,bats}: Transform empty strings as null from csv

### PR DESCRIPTION
**Pull request by katiemcculloch**

### CODE REVIEW

**Comment by katiemcculloch** [comment]
@bheni Just pushed another commit, this is ready for review again.

File: go/libraries/doltcore/type_conversion.go [hunk 1]
```diff
@@ -145,6 +151,10 @@ func convStringToFloat(val types.Value) (types.Value, error) {
 		return nil, nil
 	}
 
+	if val.Equals(types.String("\"\"")) {
```
Comment (bheni) [inline go/libraries/doltcore/type_conversion.go]: If the value in the csv is """""" then the val should be "\"\"" I don't think we want this to be Null.  It seems like that should be an error.  The same goes for the ones below.

File: go/libraries/doltcore/type_conversion_test.go [hunk 2]
```diff
@@ -42,6 +42,20 @@ func TestConv(t *testing.T) {
...
 		{types.String("true"), types.Bool(true), convStringToBool, false},
 		{types.String("anything"), types.NullValue, convToNullFunc, false},
+		{types.String(""), types.String(""), identityConvFunc, false},
+		{types.String(""), types.NullValue, convStringToUUID, false},
+		{types.String(""), types.NullValue, convStringToUint, false},
+		{types.String(""), types.NullValue, convStringToInt, false},
+		{types.String(""), types.NullValue, convStringToFloat, false},
+		{types.String(""), types.NullValue, convStringToBool, false},
+		{types.String(""), types.NullValue, convToNullFunc, false},
+		{types.String("\"\""), types.String(""), identityConvFunc, false},
```
Comment (bheni) [inline go/libraries/doltcore/type_conversion_test.go]: The expected value should be types.String("\"\"") same as what was input.

File: go/libraries/doltcore/type_conversion_test.go [hunk 3]
```diff
@@ -42,6 +42,20 @@ func TestConv(t *testing.T) {
...
+		{types.String("\"\""), types.NullValue, convStringToUUID, false},
```
Comment (bheni) [inline go/libraries/doltcore/type_conversion_test.go]: I think all of these should error

**Comment by katiemcculloch** [comment]
Re-did this work in #199

**Pull request closed without merging**